bridge/client/app/_components/ktb-webhook-settings/ktb-webhook-settings.component.ts

```diff
@@ -0,0 +1,119 @@
+import { Component, Input } from '@angular/core';
+import { AbstractControl, FormArray, FormControl, FormGroup, Validators } from '@angular/forms';
+import { DataService } from '../../_services/data.service';
+import { FormUtils } from '../../_utils/form.utils';
+import { UniformSubscription } from '../../_models/uniform-subscription';
+import { WebhookConfigMethod } from '../../../../shared/interfaces/webhook-config';
+
+type ControlType = 'method' | 'url' | 'payload' | 'proxy' | 'header';
+
+@Component({
+  selector: 'ktb-webhook-settings[subscriptionExists][projectName][subscription]',
+  templateUrl: './ktb-webhook-settings.component.html',
+  styleUrls: ['./ktb-webhook-settings.component.scss'],
+})
+export class KtbWebhookSettingsComponent {
+  public _projectName?: string;
+  public _subscription?: UniformSubscription;
+  public webhookConfigForm = new FormGroup({
+    method: new FormControl('', [Validators.required]),
+    url: new FormControl('', [Validators.required, Validators.pattern(FormUtils.URL_PATTERN)]),
+    payload: new FormControl('', []),
+    header: new FormArray([]),
+    proxy: new FormControl('', [Validators.pattern(FormUtils.URL_PATTERN)]),
+  });
+
+  public webhookMethods: WebhookConfigMethod[] = ['GET', 'POST', 'PUT'];
+  private _subscriptionExists = false;
+  public loading = false;
+
+  @Input() set subscriptionExists(status: boolean) {
+    if (this._subscriptionExists !== status) {
+      this._subscriptionExists = status;
+      this.getWebhook();
+    }
+  }
+
+  get subscriptionExists(): boolean {
+    return this._subscriptionExists;
+  }
+
+  @Input()
+  get projectName(): string {
+    return this._projectName || '';
+  }
+
+  set projectName(projectName: string) {
+    if (this._projectName !== projectName) {
+      this._projectName = projectName;
+      this.getWebhook();
+    }
+  }
+
+  @Input()
+  get subscription(): UniformSubscription | undefined {
+    return this._subscription;
+  }
+
+  set subscription(value: UniformSubscription | undefined) {
+    if (this._subscription !== value) {
+      this._subscription = value;
+      this.getWebhook();
+    }
+  }
+
+  get header(): FormArray {
+    return this.getFormControl('header') as FormArray;
+  }
+
+  get headerControls(): FormGroup[] {
+    return this.header.controls as FormGroup[];
+  }
+
+  constructor(private dataService: DataService) {
+  }
+
+  private getWebhook(): void {
+    if (this.subscriptionExists && this.subscription && this.projectName) {
```

Comment:
Instead of doing this, wouldn't it make more sense to load the webhook in the parent component, where we should be aware of all the inputs and then provide the webhook config also as an input?